### ISSUE TITLE
Output of entity changes in KIC's logs instead of `Println` in database-reconciler repo

### ISSUE THREAD

**Issue by randmonkey** [Kong/kubernetes-ingress-controller#5289]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem Statement

Currently we enable output of [`go-database-reconciler`](https://github.com/Kong/go-database-reconciler) in setup when log level is set to `trace` or `debug`:
https://github.com/Kong/kubernetes-ingress-controller/blob/64a0a6105546dd85891871345af3f25b78ec0a06/internal/manager/setup.go#L48-L50

This uses `cprint` in https://github.com/Kong/go-database-reconciler to output the entity changes, which calls https://github.com/fatih/color for printing message on the terminal in ANSI colors. This output format is suitable for terminal but not for log files. It brings inconsistency in logs, and may print special characters for colors. 

### Proposed Solution

- Fetch changes in return values of `diff.Solve`
- Output in KIC's loggers if we enable them by setting log levels

### Additional information

We should also consider updating the code in https://github.com/Kong/go-database-reconciler after separated it from decK. This repo is not mainly used as a CLI as decK, so the direct output to terminal may need to be cleaned.

### Acceptance Criteria

- [ ] When "debug" or "trace" level enabled, we can see output of entity changes (in DB mode) from KIC's logger

**Comment by randmonkey** [comment]
@rainest Could this be fixed after #5785 merged?

**Comment by rainest** [comment]
Not quite. https://github.com/Kong/kubernetes-ingress-controller/blob/5cefa9a44fe51fe09ebad0a82a7aba3b42bec2fa/internal/dataplane/sendconfig/dbmode.go#L94 is currently just discarding the list of change events from https://github.com/Kong/go-database-reconciler/blob/db51b8d3033b178fd0729c19c06429fb75ca40b3/pkg/diff/diff.go#L568 and only handling the errors.

We need to add a new log loop that does handle that with some sort of diff output, or add a debug endpoint for it. My original intent was to do the latter, since even with our own logger and without the colors, the diffs don't lend themselves well to logging because they require multiple lines to be legible, and break the usual log contract as such.

Having a endpoint that just prints the latest diff elsewhere displays that more cleanly. We could opt to still log basic "entity <whatever> changed successfully" info alongside that.

**Comment by lahabana** [comment]
@randmonkey do you think there's much to do to review and get this PR merged in? Seems like @rainest had this implemented already

**Comment by randmonkey** [comment]
OK, let me review it again if it has been already done.
